src/server/pfs/server/driver_file.go

```diff
@@ -564,7 +564,7 @@ func (d *driver) getFileSet(ctx context.Context, commit *pfsdb.Commit) (*fileset
 			// TODO: Need to handle this differently if we want to delete total
 			// file sets after a commit is finished (to save space for old commits).
 			if errors.Is(err, errNoTotalFileSet) {
-				return d.storage.Filesets.Compose(ctx, nil, defaultTTL)
+				return nil, errors.Errorf("the commit is forgotten")
```

Comment:
I'm not sure about the consequences of this.  Do empty commits have a Fileset, or are we relying on this to compose the empty list whenever we read from one?

Reply:
There is no state variable for a forgotten commit. The necessary and sufficient statement for a forgotten commit is <--> the commit is finished and has no total fileset.
To answer your questions:
1. empty commits have total fileset when they are finished.
2. When the compose is replaced by the error message, no tests were broken. So I'm not sure are there any potential use cases for this?